server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Mailbox.java

```diff
@@ -136,22 +139,22 @@ public Builder mayDelete(boolean mayDelete) {
         }
 
         public Builder totalMessages(long totalMessages) {
-            this.totalMessages = Optional.of(totalMessages);
+            this.totalMessages = Optional.of(Number.lenientFactory().fromLong(totalMessages));
```

Comment:
how do you decide sometimes lenient, sometimes strict?

Reply:
When it's parsed from the user input -> strict

When it's coming from our code -> lenient

Reply:
strange that's the contrary of the motto: be strict with what you send, be tolerant with what you get (my free interpretation)

Reply:
We can very well decide to use lenient stuff on what we receive as well. But I don't like modifying user input in the explicit protocol that JMAP is supposed to be.

Now, if we want to be strict about what we send:
 - it means we won't display ie some mailboxes when their count is negative. Or we might have similar problems with similar values.
 - we anyway need to push the sanitizing lower in the stack, IMO to the Dao